certificates_test.go

```diff
@@ -0,0 +1,186 @@
+package ydb
+
+import (
+	"context"
+	"os"
+	"sync"
+	"testing"
+
+	"github.com/ydb-platform/ydb-go-sdk/v3/internal/certificates"
+	"github.com/ydb-platform/ydb-go-sdk/v3/internal/conn"
+)
+
+func BenchmarkWithCertificateCache(b *testing.B) {
+	b.ReportAllocs()
+
+	bytes, err := os.ReadFile(os.Getenv("YDB_SSL_ROOT_CERTIFICATES_FILE"))
```

Comment:
I could add a `testdata/` directory and put a root CAs file from the current version of ydb container there. This will also improve the reproducibility of tests/benchmarks, which is also good.

Reply:
ydb container generate new tmp certificates on every create.

If you will put fixed certificate to fixed path - it must have a long time before expire (100 years for example).

Reply:
Now the benchmarks don't actually connect to YDB, so for them the expiration date of certificate doesn't really matter. The tests, however, do connect to the YDB, but it also means that they require additional work in a form of starting YDB container (like other e2e tests in this project).

I could rewrite the tests like benchmarks, but I'm afraid it might make concurrency test less informative (we won't check for concurrent cert access during concurrent child connections to database).